api/xds_protocol.rst

```diff
@@ -613,6 +613,43 @@ adding/removing/updating clusters. On the other hand, routes are not
 warmed, i.e., the management plane must ensure that clusters referenced
 by a route are in place, before pushing the updates for a route.
 
+.. _xds_protocol_TTL:
+
+TTL
+~~~
+
+In the event that the management server becomes unreachable, the last known configuration received
+by Envoy will persist until the connection is reestablished. For some services, this may not be
+desirable. For example, in the case of a fault injection service, a management server crash at the
+wrong time may leave Envoy in an undesirable state. The TTL setting allows Envoy to remove a set of
+resources after a specified period of time if contact with the management server is lost. This can
+be used, for example, to terminate a fault injection test when the management server can no longer
+be reached.
+
+SotW TTL
+^^^^^^^^
+
+A TTL field in the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>` specifies the TTL for
+all resources contained in the response. A timer is started when the response is received. When the
+timer expires, all resources retrieved from the management server will be removed.
+
+The TTL timer can be updated by sending a specially formed 
+:ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>` with the same version as the previously
```

Comment:
Re-using the same version for a new response means that the client would not have any way to ACK or NACK the TTL change.  This seems sub-optimal.

Why not just re-send the resources when their TTLs change?  This doesn't seem like something that will happen that often.

Reply:
Agreed. @htuch suggested replacing this with a 'refresh_ttl' boolean field. I have some questions about that approach that I asked in another part of this PR. I do like your proposal of just resending the resources to reset the TTL. I think this is simpler @mattklein123 @htuch what do you think?

Reply:
My concern around re-sending resources is that it seems pretty bad for perf if we need to serialize/send a bunch of proto each time we want to refresh the TTL. With that said, that might be premature optimization. I think it would be fine to start with this and we can always add a "refresh" type field later if we really need it?

Reply:
I think that makes sense